src/perform_step/extrapolation_perform_step.jl

```diff
@@ -230,6 +230,101 @@ function perform_step!(integrator,cache::AitkenNevilleConstantCache,repeat_step=
   integrator.k[2] = integrator.fsallast
 end
 
+
+function initialize!(integrator,cache::ImplicitEulerExtrapolationConstantCache)
+  integrator.kshortsize = 2
+  integrator.k = typeof(integrator.k)(undef, integrator.kshortsize)
+  integrator.fsalfirst = integrator.f(integrator.uprev, integrator.p, integrator.t) # Pre-start fsal
+
+  # Avoid undefined entries if k is an array of arrays
+  integrator.fsallast = zero(integrator.fsalfirst)
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+end
+
+function perform_step!(integrator,cache::ImplicitEulerExtrapolationConstantCache,repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack dtpropose, T, cur_order, work, A, nlsolver = cache
+
+  alg = unwrap_alg(integrator, true)
+  max_order = min(size(T)[1], cur_order+1)
+
+  for i in 1:max_order
+    dt_temp = dt/(2^(i-1)) # Romberg sequence
+    u = uprev
+    fsal_first = integrator.fsalfirst
+    for j in 1:2^(i-1)
+        integrator.dt = dt_temp
+        update_W!(integrator, cache, dt_temp, repeat_step)
+
+        # initial guess
+        if alg.extrapolant == :linear
+          nlsolver.z = dt_temp*fsal_first
```

Comment:
We should get much better interpolants in most cases here. For example, if the `j=1` case has been calculated, we can just use a linear interpolation between t and t+dt values.